### PR TITLE
fix: log out users that are not partners

### PR DESCRIPTION
**Pull request by pvinis**
### Description

This PR kinda resolves [MOPLAT-425]

When logging in with a user that is not a partner, the app would error out (crash in beta). Now it will instead just log you out safely.

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-425]: https://artsyproduct.atlassian.net/browse/MOPLAT-425?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/navigation/Main.tsx [hunk 1]
```diff
@@ -201,3 +201,9 @@ export const Main = () => {
     </NavigationContainer>
   )
 }
+
+const ErrorBoundaryWrapper = (Component: FC) => (
```
Comment (pvinis) [inline src/app/navigation/Main.tsx:205]: i moved this here since its only to be used in this file.

File: src/app/wrappers/ErrorBoundary.tsx [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-  <ErrorBoundary>
-    <Component />
-  </ErrorBoundary>
-)
-
 // Taken from https://relay.dev/docs/guided-tour/rendering/error-states/#when-using-uselazyloadquery
 interface ErrorBoundaryProps {
   children: React.ReactNode
   withoutBackButton?: boolean
+  catch?: (error: Error) => void
```
Comment (pvinis) [inline src/app/wrappers/ErrorBoundary.tsx:8]: a nice little prop that we can run some code in there when the error is caught.

File: src/app/screens/Auth/SelectPartner.tsx [hunk 3]
```diff
@@ -28,9 +29,18 @@ export const SelectPartnerHeader = ({ onSearchChange, searchValue }: SelectPartn
 export const SelectPartnerScreen = () => (
   <Screen>
     <Screen.Body>
-      <Suspense fallback={<ActivityIndicator />}>
-        <SelectPartner />
-      </Suspense>
+      <ErrorBoundary
+        catch={(e) => {
```
Comment (pvinis) [inline src/app/screens/Auth/SelectPartner.tsx:33]: boom!

**Review by rajsam003 — APPROVED** [review]
💪

**Pull request merged**